test/service_test.cc

```diff
@@ -130,24 +130,27 @@ class ServiceTestWithParameterizedConstructor : public ServiceTest {
     setBasicRequestOptions();
   }
 
+private:
   Envoy::Thread::MutexBasicLockable log_lock_;
 };
 
 INSTANTIATE_TEST_SUITE_P(IpVersions, ServiceTestWithParameterizedConstructor,
                          ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()),
                          Envoy::TestUtility::ipTestParamsToString);
 
-// Test single NH run with parameterized constructor
-TEST_P(ServiceTestWithParameterizedConstructor, BasicWithUserSpecifiedLoggingContext) {
-  auto r = stub_->ExecutionStream(&context_);
-  r->Write(request_, {});
-  r->WritesDone();
-  EXPECT_TRUE(r->Read(&response_));
+TEST_P(ServiceTestWithParameterizedConstructor,
+       ConstructorWithLoggingContextParameterCanRespondToRequests) {
+  std::unique_ptr<::grpc::ClientReaderWriter<::nighthawk::client::ExecutionRequest,
+                                             ::nighthawk::client::ExecutionResponse>>
+      request = stub_->ExecutionStream(&context_);
```

Comment:
Ah, I actually gave you a bad name for this. This isn't actually a request.  Sorry about that!  Easier to tell what it is now that it's not an `auto`.

Whatever makes sense to you as a name here.  reader_writer is an option.  Or if you don't like that, something that describes this accurately, like stream_client or stream_handler.  Whatever you feel good about.

Reply:
I would go with stream. Thanks for the advice!